### PR TITLE
ci: s3tests runner collect more data

### PR DESCRIPTION
**Pull request by m-ildefons**
- Add a bit of metadata to the report json during nightly runs to help identify what was tested

- Collect logs in the report json during s3tests for later consumption in the web UI of the s3gw status repo

- Add timeout to the s3tests, avoiding hanging tests from messing with CI

- Enable running all s3tests to collect overview of what failed too

### CODE REVIEW

File: tools/tests/s3tests-runner.sh [hunk 1]
```diff
@@ -187,22 +188,33 @@ _run() {
...
-    ".tests += [{\"name\": \"${name}\", \"result\": \"${result}\", \"slot\": \"${slot}\", \"time\": \"$runtime\"}]" \
-    "${TMPFILE}" ; ) 255>"${TMPFILE}.lock"
+  if [ "$CONTAINER_CMD" = "docker" ] ; then
+    docker logs "$CONTAINER" > "${OUTPUT_DIR}/logs/${test}/radosgw.log" 2>&1
+  elif [ -n "$CONTAINER" ] ; then
+    mv "${OUTPUT_DIR}/logs/radosgw.log" "${OUTPUT_DIR}/logs/${test}/radosgw.log"
+  fi
+
+
+  ( flock -w 5 255 ; \
```
Comment (irq0) [inline tools/tests/s3tests-runner.sh]: Why share TMPFILE between processes? Could this write into separate files and have a combine step shortly before _convert? 
 
TMPFILE could use a better name that includes what is in there.

File: tools/tests/s3tests-runner.sh [hunk 2]
```diff
@@ -175,7 +176,7 @@ _run() {
 
   export S3TEST_CONF="${TMPDIR}/s3tests.conf"
   export S3_USE_SIGV4=ON
-  if pytest \
+  if timeout --preserve-status "${S3TEST_TIMEOUT}" pytest \
```
Comment (irq0) [inline tools/tests/s3tests-runner.sh]: Would it make sense to add a result "timeout"?

**Review by irq0 — CHANGES_REQUESTED** [review]

File: tools/tests/s3tests-runner.sh (hunk 1 above)
Comment (m-ildefons) [reply tools/tests/s3tests-runner.sh]: Done. Collect results in separate files, merge files in the `_convert` step

File: tools/tests/s3tests-runner.sh (hunk 2 above)
Comment (m-ildefons) [reply tools/tests/s3tests-runner.sh]: Yes, done.

**Comment by m-ildefons** [comment]
Again, the nightly pipeline here can't succeed because it's a PR from a fork.

**Review by jecluis — APPROVED** [review]

**Comment by m-ildefons** [comment]
Closed in favor of: https://github.com/aquarist-labs/s3gw/pull/528

**Pull request closed without merging**